src/include/daos/mem.h

```diff
@@ -632,4 +632,129 @@ umem_tx_add_callback(struct umem_instance *umm, struct umem_tx_stage_data *txd,
 	D_ASSERT(umm->umm_ops->mo_tx_add_callback != NULL);
 	return umm->umm_ops->mo_tx_add_callback(umm, txd, stage, cb, data);
 }
+
+/*********************************************************************************/
+
+/** Describing a storage region for I/O */
+struct umem_store_region {
+	/** start offset of the region */
+	daos_off_t	sr_addr;
```

Comment:
Though daos_off_t and umem_off_t are both 64 bits integer, several bits of umem_off_t are reserved for flags. It's better to use umem_off_t here.